### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by jpuri**
* Code cleanup
* Adding detailed comments

### CODE REVIEW

File: src/ppom-controller.test.ts [hunk 1]
```diff
@@ -114,21 +114,21 @@ describe('PPOMController', () => {
         return Promise.resolve();
       });
       jest.runOnlyPendingTimers();
-      expect(spy).toHaveBeenCalledTimes(7);
+      expect(spy).toHaveBeenCalledTimes(6);
```
Comment (segun) [inline src/ppom-controller.test.ts:117]: I think I've seen these values change in a couple of PRs. I'll like to understand why the values keep changing and if there's something to do to make sure we don't have new PRs every time they changed

File: src/ppom-controller.ts [hunk 2]
```diff
@@ -379,60 +389,53 @@ export class PPOMController extends BaseControllerV2<
...
 
   /**
-   * Determine if an update to the ppom configuration is needed.
-   * The function will return true if data is not already fetched for the chain.
-   *
-   * @returns True if PPOM data requires update.
+   * The function will return true if data is not already fetched for current chain.
    */
-  async #shouldUpdate(): Promise<boolean> {
+  async #isDataRequiredForCurrentChain(): Promise<boolean> {
```
Comment (cryptotavares) [inline src/ppom-controller.ts:417]: No need for it to be an async method

File: src/ppom-controller.ts [hunk 3]
```diff
@@ -702,39 +699,51 @@ export class PPOMController extends BaseControllerV2<
     return response;
   }
 
+  /*
+   * Function sends a HEAD request to version info file and compares the ETag to the one saved in controller state.
+   * If ETag is not changed we can be sure that there is not change in files and we do not need to fetch data again.
+   */
+  async #versionInfoETagChanged(url: string): Promise<boolean> {
```
Comment (cryptotavares) [inline src/ppom-controller.ts:706]: This method name can be improved.. maybe `checkIfVersionInfoETagChanged`? Although it also updates the controller state if the etag has changed. So maybe something else?

File: src/ppom-controller.ts [hunk 4]
```diff
@@ -327,16 +290,13 @@ export class PPOMController extends BaseControllerV2<
   /**
    * Update the PPOM.
    * This function will acquire mutex lock and invoke internal method #updatePPOM.
-   *
-   * @param options - Options.
-   * @param options.updateForAllChains - True is update if required to be done for all chains in cache.
    */
-  async updatePPOM({ updateForAllChains } = { updateForAllChains: true }) {
+  async updatePPOM() {
```
Comment (cryptotavares) [inline src/ppom-controller.ts:294]: why is this method public? Do we intend to use it on either the extension or mobile?

Comment (cryptotavares) [reply src/ppom-controller.ts:294]: Nevermind.. saw that it is being exposed through the message handlers

Comment (jpuri) [reply src/ppom-controller.ts:294]: Yeh it is not currently being used outside the controller, we can make it private also. Let me do that.

File: src/ppom-controller.test.ts (hunk 1 above)
Comment (jpuri) [reply src/ppom-controller.test.ts:121]: I actually did a refactoring in this PR to reduce number of call to version info file when `usePPOM` is called. That resulted in this number of calls getting reduced.

**Review by segun — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**